apps/gyrokinetic_update_ssp_rk3.c

```diff
@@ -216,17 +216,21 @@ gyrokinetic_update_ssp_rk3(gkyl_gyrokinetic_app* app, double dt0)
         else {
           struct timespec wst = gkyl_wall_clock();
           for (int i=0; i<app->num_species; ++i) {
-	    struct gk_species *gks = &app->species[i];
+	          struct gk_species *gks = &app->species[i];
             // Step f.
-	    gk_species_combine(gks, gks->f1, 1.0/3.0, gks->f, 2.0/3.0, gks->fnew, &gks->local_ext);
-	    gk_species_copy_range(gks, gks->f, gks->f1, &gks->local_ext);
+            gk_species_combine(gks, gks->f1, 1.0/3.0, gks->f, 2.0/3.0, gks->fnew, &gks->local_ext);
+            gk_species_copy_range(gks, gks->f, gks->f1, &gks->local_ext);
             // Step boundary fluxes.
             gk_species_bflux_combine(app, &gks->bflux, gks->bflux.f1,
               1.0/3.0, gks->bflux.f, 2.0/3.0, gks->bflux.fnew);
             gk_species_bflux_copy(app, &gks->bflux, gks->bflux.f, gks->bflux.f1);
             gk_species_bflux_calc_voltime_integrated_mom(app, gks, &gks->bflux, tcurr);
             gk_species_bflux_scale(app, &gks->bflux, gks->bflux.f, 1.0/dt);
+
+            // adapt the sources
+            gk_species_source_adapt(app, gks, &gks->src, gks->lte.f_lte, tcurr);
```

Comment:
@Antoinehoff please move this back down to where it was before (I think) because the problem with the new location is that you included it inside the `time_stepper_arithmetic_tm` time, but this is incorrect, we do not want to include the time spent in source_adapt in this time.